setuptools/config/setupcfg.py

```diff
@@ -174,6 +175,41 @@ def parse_configuration(
     return meta, options
 
 
+def warn_about_potential_extras_misconfig(section_options, parsed):
```

Comment:
`extras` might be a bit confusing here because it could also refer to the `extra` environment marker used inside the requirement. How about `warn_accidental_extras_require_misconfig` (or `verify_accidental_extras_require_misconfig`)?